app/post-contribution/route.js

```diff
@@ -17,7 +17,9 @@ export default Route.extend(AuthenticatedRouteMixin, {
   model({ date = moment().format('YYYY-MM-DD') }) {
     let user = this.get('currentUser.user')
 
-    return this.store.createRecord('contribution', { user, date })
+    let challenge = this.store.findRecord('challenge', 'current')
```

Comment:
Do you think? In the edit-route, a real existing contribution is loaded, which should have a reference to the challenge, which should therefore be followed by ember automatically. Not sure yet though, and edit-contribution tests are missing a bit :/